extensions/wikia/ArticleAsJson/ArticleAsJson.class.php

```diff
@@ -322,4 +450,57 @@ private static function isIconSize( $sizeParam ) {
 	private static function isInfoIcon( $templateType ) {
 		return $templateType == TemplateClassificationService::TEMPLATE_INFOICON;
 	}
+
+	/**
+	 * @param $originalHeight
+	 * @param $originalWidth
+	 *
+	 * @return array
+	 */
+	private static function scaleIconSize( $originalHeight, $originalWidth ) {
+		$height = $originalHeight;
+		$width = $originalWidth;
+		$maxHeight = self::ICON_SCALE_TO_MAX_HEIGHT;
+
+		if ( $originalHeight > $maxHeight ) {
+			$height = $maxHeight;
+			$width = intval( $maxHeight * $originalWidth / $originalHeight );
+		}
+
+		return [
+			'height' => $height,
+			'width' => $width
+		];
+	}
+
+	private static function getMediaDetailWithSizeFallback(
+		$title, $mediaDetailConfig, $fallbackSize=self::MAX_MERCURY_IMAGE_WIDTH
+	) {
+		$mediaDetail = WikiaFileHelper::getMediaDetail( $title, $mediaDetailConfig );
+		if ( $mediaDetail['width'] == 0 ) {
```

Comment:
If we don't want to use `===` (kinda makes sense) then maybe this should be `if ( empty(...) )`?
http://php.net/manual/en/function.empty.php